scripts/automatic_tag.go

```diff
@@ -0,0 +1,246 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/transport/ssh"
+	"golang.org/x/mod/semver"
+)
+
+const (
+	sdkRepo      = "git@github.com:stackitcloud/stackit-sdk-go.git"
+	patch        = "patch"
+	minor        = "minor"
+	coreOnlyFlag = "--core-only"
+)
+
+var (
+	updateTypes = []string{minor, patch}
+	usage       = fmt.Sprintf("go run automatic_tag.go [minor|patch] ssh-private-key-file-path password %s", coreOnlyFlag)
+)
+
+func main() {
+	// Parse arguments
+	if len(os.Args) != 4 && len(os.Args) != 5 {
+		fmt.Printf("wrong number of arguments. Usage: %s\n", usage)
+		os.Exit(1)
+	}
+
+	updateType := os.Args[1]
+	valid := false
+	for _, t := range updateTypes {
+		if updateType == t {
+			valid = true
+			break
+		}
+	}
+	if !valid {
+		fmt.Printf("the provided argument for update type `%s` is not valid, the valid values are: [%s]\n", updateType, strings.Join(updateTypes, ","))
+		os.Exit(1)
+	}
+
+	sshPrivateKeyFilePath := os.Args[2]
+	_, err := os.Stat(sshPrivateKeyFilePath)
+	if err != nil {
+		fmt.Printf("The provided private key file path %s is not valid: %s\nUsage: %s\n", sshPrivateKeyFilePath, err, usage)
+		os.Exit(1)
+	}
+
+	sshKeyPassword := os.Args[3]
+	if sshKeyPassword == coreOnlyFlag {
+		fmt.Printf("Wrong arguments. Usage: %s\n", usage)
+		os.Exit(1)
+	}
+
+	// Parse flag
+	var coreOnly bool
+	if len(os.Args) == 5 {
+		if os.Args[4] != coreOnlyFlag {
+			fmt.Printf("Wrong arguments. Usage: %s\n", usage)
+			os.Exit(1)
+		}
+		coreOnly = true
+	}
+
+	err = automaticTagUpdate(updateType, sshPrivateKeyFilePath, sshKeyPassword, coreOnly)
+	if err != nil {
+		fmt.Printf("Error updating tags: %s\n", err.Error())
+		os.Exit(1)
+	}
+}
+
+func automaticTagUpdate(updateType, sshPrivateKeyFilePath, password string, coreOnly bool) error {
+	tempDir, err := os.MkdirTemp("", "")
+	if err != nil {
+		return fmt.Errorf("create temporary directory: %w", err)
+	}
+
+	defer func() {
+		tempErr := os.RemoveAll(tempDir)
+		if tempErr != nil && err == nil {
+			err = fmt.Errorf("temporary directory %s could not be removed: %w", tempDir, tempErr)
```

Comment:
I think for this to work you would need to have a named return variable:
```go
func automaticTagUpdate(...) (err error)
```

Reply:
Because this would just set the variable `err` defined in the first line of the function, but it is never returned

Reply:
Alternatively we could just print a message